pkgs/os-specific/linux/minimal-bootstrap/stage0-posix/default.nix

```diff
@@ -0,0 +1,36 @@
+{ lib
+, system
+, newScope
+}:
+
+lib.makeScope newScope (self: with self; {
+  callPackage = self.callPackage;
+
+  # Pinned from https://github.com/oriansj/stage0-posix/commit/bdd3ee779adb9f4a299059d09e68dfedecfd4226
+  version = "unstable-2023-04-24";
+
+  # We don't have access to utilities such as fetchgit and fetchzip since they
+  # would introduce a circular dependency. The only tool we have to fetch source
+  # trees is `import <nix/fetchurl.nix>` with the unpack option, taking a
```

Comment:
Why does this approach avoid the circular dependency? Is it because it's implemented in nix? Could be good to explain this in the comment.